src/main/python/mlsearch/protocol.py

```diff
@@ -0,0 +1,61 @@
+class Protocol():
+    """The Protocol for standard communication accross different api sources."""
+
+    def __init__(self, kwargs):
+        param_list = [
+            
+            # title -> paper_title, full_name, name
+            # description -> paper_abstract, description
+
+            # Paper with code
+            'title',
+            'paper_published', 'paper_url',
+            'num_of_implementations', 'tasks',
+            'paper_conference', 'repository_url',
+            'repository_name', 'repository_framework',
+            'repository_stars',
+            
+            # Github
+            'description', 'private',
+            'fork', 'updated_at',
+            'stargazers_count', 'watchers_count',
+            'language', 'forks_count',
+            
+            # Coursera
+            'partners_v1', 'instructors_v1',
+            
+            # Source Flag
+            'source'
+        ]
+        
+        for param in kwargs:
+            if param not in param_list:
+                raise AttributeError('{} is not a valid parameter.'.format(param))
+        
+        self.title = kwargs.get('title', None)
```

Comment:
it's s just my concern, CMIIW. I'm wondering what happened if there's no `title` and `None` is passed as the response. since it has a different type compared to an empty string or null.

Reply:
I confirmed it on json format and `None` is converted to `null`. I guess the client side need to confirm this behavior on their respective programming platform.